### DISCUSSION TITLE
Image processing

### DISCUSSION THREAD

**Discussion by sim186** [Beep6581/RawTherapee#6492] [Q&A]
Can someone address me where (if is this the case) are located the code of the shaders used from RawTherapee for the image processing (exposure, brightness, tone, etc...)? I would like to use something similar for my other project. 

Thanks

**Comment by Lawrence37** [comment]
They are in the rtengine directory. There are many of them, so it might be easier to find the name of the parameter you want in ProcParams and search for it in the code.
